### DISCUSSION TITLE
MosaicJSON 424 error with http asset urls

### DISCUSSION THREAD

**Discussion by scottyhq** [developmentseed/titiler#736] [Q&A] [answered]
I noticed some odd behavior with the default `/mosaicjson` endpoint with 424 (Failed Dependency) errors if the mosaic.json contains HTTP links.

You can reproduce the error by modifying the mosaic JSON in this example to use http urls (`s3://noaa-eri-pds --> https://noaa-eri-pds.s3.amazonaws.com`)
https://developmentseed.org/titiler/examples/notebooks/Working_with_MosaicJSON/

For example, this errors:
https://titiler.xyz/mosaicjson/map?url=https://gist.github.com/scottyhq/4c288a2dbf7b97ef7833c5e0ee19d6d6/raw/636083cf7117a96e38d271eeef932a3ee507d7d1/NOAA_Nashville_Tornado_v2_http.json

I see the same error running a local 0.15.6 titiler, but don't see any logs beyond 424 (Failed Dependency) 
http://localhost:8000/mosaicjson/map?url=https://gist.github.com/scottyhq/4c288a2dbf7b97ef7833c5e0ee19d6d6/raw/636083cf7117a96e38d271eeef932a3ee507d7d1/NOAA_Nashville_Tornado_v2_http.json

Curiously if I use the mosaic.json from a local path it works!
http://localhost:8000/mosaicjson/map?url=file:///Users/scott/Downloads/NOAA_Nashville_Tornado_v2_http.json

https://github.com/developmentseed/titiler/discussions/644 could be related in terms of some difference in how credentials are assumed in `/mosaicjson` versus `/cog` endpoints. The bucket is fully public and picking any single COG http url from the mosaic works with the `/cog/map` endpoint:
https://titiler.xyz/cog/map?url=https://noaa-eri-pds.s3.amazonaws.com/2020_Nashville_Tornado/20200307a_RGB/20200307aC0862230w361030n.tif

**Comment by vincentsarago** [comment]
@scottyhq 

> For example, this errors:
https://titiler.xyz/mosaicjson/map?url=https://gist.github.com/scottyhq/4c288a2dbf7b97ef7833c5e0ee19d6d6/raw/636083cf7117a96e38d271eeef932a3ee507d7d1/NOAA_Nashville_Tornado_v2_http.json

From what I can see, the `tilejson` endpoint is failing with 

```
{
  "detail": "b''"
}
```
I think the issue comes from the gist url that gets redirected somehow! 

I just tried with `https://gist.githubusercontent.com/scottyhq/4c288a2dbf7b97ef7833c5e0ee19d6d6/raw/636083cf7117a96e38d271eeef932a3ee507d7d1/NOAA_Nashville_Tornado_v2_http.json`

`https://titiler.xyz/mosaicjson/map?url=https://gist.githubusercontent.com/scottyhq/4c288a2dbf7b97ef7833c5e0ee19d6d6/raw/636083cf7117a96e38d271eeef932a3ee507d7d1/NOAA_Nashville_Tornado_v2_http.json` and it worked 🤷

**Reply by scottyhq** [reply, marked as answer]
huh, good catch! So the 'raw' button link in either github gist or repository results in the path with a direct that doesn't work. Instead use the final redirected link: 

For example:
https://github.com/developmentseed/mosaicjson-spec/blob/main/0.0.2/example/dg_post_idai.json

DONT USE: https://github.com/developmentseed/mosaicjson-spec/raw/main/0.0.2/example/dg_post_idai.json

USE:
https://raw.githubusercontent.com/developmentseed/mosaicjson-spec/main/0.0.2/example/dg_post_idai.json

(https://titiler.xyz/mosaicjson/map?url=https://raw.githubusercontent.com/developmentseed/mosaicjson-spec/main/0.0.2/example/dg_post_idai.json)
